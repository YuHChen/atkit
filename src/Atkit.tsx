import React from "react";
import { Container, Nav, Navbar } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { HashRouter as Router, Route, Routes } from "react-router-dom";

import logo from "./assets/logo.png";

import Cards from "./cards/Cards";
import Combos from "./combos/Combos";
import ComboMap from "./deck/ComboMap";
import Home from "./home/Home";
import NoMatch from "./NoMatch";

import { CardsDataDumper, CombosDataDumper } from "./devtools/dataDumpers";

interface DevoAwareProps {
  isDevo: boolean;
}

const Logo = () => (
  <LinkContainer to="/">
    <Navbar.Brand>
      <img
        src={logo}
        width="30"
        height="30"
        className="d-inline-block align-top"
        alt="atkit logo"
      />
    </Navbar.Brand>
  </LinkContainer>
);

const NavBar = (props: DevoAwareProps) => (
  <Navbar collapseOnSelect expand="sm" bg="dark" variant="dark" sticky="top">
    <Container fluid>
      <Logo />

      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        {/* Set activeKey="/" to mark nav links as inactive when on home page*/}
        <Nav variant="pills" activeKey="/">
          <LinkContainer to="/cards">
            <Nav.Link>Cards</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/combos">
            <Nav.Link>Combos</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/combomap">
            <Nav.Link>Combo Map</Nav.Link>
          </LinkContainer>
          {props.isDevo && (
            <LinkContainer to="/dumper/cards">
              <Nav.Link>Cards Data Dumper</Nav.Link>
            </LinkContainer>
          )}
          {props.isDevo && (
            <LinkContainer to="/dumper/combos">
              <Nav.Link>Combos Data Dumper</Nav.Link>
            </LinkContainer>
          )}
        </Nav>
      </Navbar.Collapse>
    </Container>
  </Navbar>
);

const Atkit = (props: DevoAwareProps) => (
  <Router>
    <div className="atkit">
      <NavBar isDevo={props.isDevo} />
      <div className="mt-3" />
      <Container>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/cards" element={<Cards />} />
          <Route path="/combos" element={<Combos />} />
          <Route path="/combomap" element={<ComboMap />} />
          {props.isDevo && (
            <Route path="/dumper/cards" element={<CardsDataDumper />} />
          )}
          {props.isDevo && (
            <Route path="/dumper/combos" element={<CombosDataDumper />} />
          )}
          <Route path="*" element={<NoMatch />} />
        </Routes>
      </Container>
    </div>
  </Router>
);

export default Atkit;
