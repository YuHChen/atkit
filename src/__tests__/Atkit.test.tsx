import React from "react";
import { render } from "@testing-library/react";

import Atkit from "../Atkit";

describe("Atkit", () => {
  test("renders as expected in production", () => {
    const { container } = render(<Atkit isDevo={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders as expected in development", () => {
    const { container } = render(<Atkit isDevo={true} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
