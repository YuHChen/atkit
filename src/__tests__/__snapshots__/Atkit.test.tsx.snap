// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Atkit > renders as expected in development 1`] = `
<div
  class="atkit"
>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark sticky-top"
  >
    <div
      class="container-fluid"
    >
      <a
        class="active navbar-brand"
        href="#/"
      >
        <img
          alt="atkit logo"
          class="d-inline-block align-top"
          height="30"
          src="/atkit/src/assets/logo.png"
          width="30"
        />
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="navbar-nav nav-pills"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="#/cards"
            href="#/cards"
          >
            Cards
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/combos"
            href="#/combos"
          >
            Combos
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/combomap"
            href="#/combomap"
          >
            Combo Map
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/dumper/cards"
            href="#/dumper/cards"
          >
            Cards Data Dumper
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/dumper/combos"
            href="#/dumper/combos"
          >
            Combos Data Dumper
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="mt-3"
  />
  <div
    class="container"
  >
    <div>
      <div
        class="p-5 mb-4 bg-body-tertiary rounded-3"
      >
        <h1
          class="display-5"
        >
          Welcome to atkit!
        </h1>
        <p>
          atkit is a toolkit for
           
          <a
            class="text-decoration-none"
            href="http://www.animationthrowdowngame.com"
          >
            Animation Throwdown: The Quest for Cards
          </a>
           
          game.
        </p>
      </div>
      <div
        class="fade alert alert-info show"
        role="alert"
      >
        <div
          class="alert-heading h4"
        >
          Pardon the mess
        </div>
        <p
          class="mb-0"
        >
          This 
          website
           is still a work in progress.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Atkit > renders as expected in production 1`] = `
<div
  class="atkit"
>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark sticky-top"
  >
    <div
      class="container-fluid"
    >
      <a
        class="active navbar-brand"
        href="#/"
      >
        <img
          alt="atkit logo"
          class="d-inline-block align-top"
          height="30"
          src="/atkit/src/assets/logo.png"
          width="30"
        />
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="navbar-nav nav-pills"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="#/cards"
            href="#/cards"
          >
            Cards
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/combos"
            href="#/combos"
          >
            Combos
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="#/combomap"
            href="#/combomap"
          >
            Combo Map
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="mt-3"
  />
  <div
    class="container"
  >
    <div>
      <div
        class="p-5 mb-4 bg-body-tertiary rounded-3"
      >
        <h1
          class="display-5"
        >
          Welcome to atkit!
        </h1>
        <p>
          atkit is a toolkit for
           
          <a
            class="text-decoration-none"
            href="http://www.animationthrowdowngame.com"
          >
            Animation Throwdown: The Quest for Cards
          </a>
           
          game.
        </p>
      </div>
      <div
        class="fade alert alert-info show"
        role="alert"
      >
        <div
          class="alert-heading h4"
        >
          Pardon the mess
        </div>
        <p
          class="mb-0"
        >
          This 
          website
           is still a work in progress.
        </p>
      </div>
    </div>
  </div>
</div>
`;
